Feature/legendary support

dyyd opened the pull request:
Added option to install from legendary (https://github.com/derrod/legendary).

Allows to run EGS games with online support through Steam without opening EGS through URI call.

This should make it easier to use in Linux as well as legendary was initially meant for Linux usage.

dyyd commented:
1. Sure, will do
2. Not yet, don't have games on my Linux machine
3. I simply didn't get the GUI working. I even tried building it in dedicated docker images but still no luck.  Reverted for now until I can actually run it and validate stuff.

Developing on Linux or WSL  and playing on Windows resulted in situation where the dev and use machines don't match.

I use legendary mainly to get account auth and drm working through Steam Link without the need to open up Epic Store.

jaydenmilne commented on steamsync-library/src/steamsync.py:
```diff
@@ -121,6 +127,45 @@ def parse_arguments():
+    """
+    games_dict = {}
+    # populate info for all installable games
+    games_raw_json = subprocess.Popen([executable_cmd, "list-games", "--json"], stdout=subprocess.PIPE).communicate()[0].decode()
```
heh this is easy mode compared to the other launchers

jaydenmilne replied:
Huh so the list games command has output the list-installed command doesn't?

dyyd replied:
Yeah. The output for installed games is quite short in comparison and only provides information required for (downloading and) running the executable basically. 
The general games list output contains all metadata Epic provides, it seems. Including cover art, DLC data etc.

jaydenmilne commented on steamsync-library/src/steamsync.py:
```diff
@@ -121,6 +127,45 @@ def parse_arguments():
+                install_location,
+                "",
+                art_url,
+                defs.TAG_EPIC,
```
Can you make a new storetag for legendary?

dyyd replied:
I can switch this to the legendary tag yes. I initially left it like this for URI usage but it makes sense to distinguish them.

jaydenmilne commented:
One final thing, can you put legendary_collect_games in its own file? I'm trying to keep the non-egs collectors out of steamsync.py, Other than that, a few minor nits and this LGTM!

Thanks for contributing!

dyyd commented:
Updated README, moved legendary collection function to separate file.
Also made a few improvements to work in Linux.

If the empty shortcuts file change is too much for this MR then I can break it off into a separate one.

jaydenmilne commented:
I’ve been very busy with some things in my personal life recently. I appreciate your contribution and will try and review it once the dust settles. Sorry for the delay.

> On Feb 15, 2022, at 02:58, Tiit Kass ***@***.***> wrote:
> 
> ﻿
> Updated README, moved legendary collection function to separate file.
> Also made a few improvements to work in Linux.
> 
> If the empty shortcuts file change is too much for this MR then I can break it off into a separate one.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you commented.
